modules/pam_faillock/tst-pam_faillock-retval.c

```diff
@@ -0,0 +1,115 @@
+/*
+ * Check pam_faillock return values.
+ */
+
+#include "test_assert.h"
+
+#include <limits.h>
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+#include <security/pam_appl.h>
+
+#define MODULE_NAME "pam_faillock"
+#define TEST_NAME "tst-" MODULE_NAME "-retval"
+
+static const char service_file[] = TEST_NAME ".service";
+static const char config_filename[] = TEST_NAME ".conf";
+static const char user_name[] = "root";
+static struct pam_conv conv;
+
+int
+main(void)
+{
+	pam_handle_t *pamh = NULL;
+	FILE *fp;
+	char cwd[PATH_MAX];
+
+	ASSERT_NE(NULL, getcwd(cwd, sizeof(cwd)));
+
+	ASSERT_NE(NULL, fp = fopen(config_filename, "w"));
+	ASSERT_LT(0, fprintf(fp,
+                  "deny = 2\n"
+                  "unlock_time = 5\n"
+                  "root_unlock_time = 5\n"));
+	ASSERT_EQ(0, fclose(fp));
+
+	/* root has access */
+	ASSERT_NE(NULL, fp = fopen(service_file, "w"));
+	ASSERT_LT(0, fprintf(fp, "#%%PAM-1.0\n"
+		  "auth required %s/../pam_permit/.libs/pam_permit.so\n"
+		  "auth required %s/.libs/%s.so authsucc even_deny_root dir=%s conf=%s\n"
+		  "account required %s/.libs/%s.so dir=%s\n",
+		  cwd,
+		  cwd, MODULE_NAME, cwd, config_filename,
+	          cwd, MODULE_NAME, cwd));
```

Comment:
Something went wrong with the indentation here.

Reply:
hopefully fixed :-)